pkg/stagemanager/stage_manager.go

```diff
@@ -322,7 +318,7 @@ func (stm *StageManager) WaitFinish() {
 // graceful stop handlers,
 // will exit with non-zero code when a callback handler return error
 func (stm *StageManager) AppendGracefulStopStage(f func(Application) error) *StageManager {
-	if f == nil || stm.started {
+	if f == nil || stm.state == Running {
 		log.StartLogger.Errorf("[stage] invalid stage function or already started")
```

Comment:
```suggestion
	if f == nil || stm.state > Running {
		log.StartLogger.Errorf("[stage] invalid stage function or already stopping")
```
这里可以放开一些，允许 Running 状态也追加，更好一些~
不过，同时也需要给 `gracefulStopStages` 加并发锁，避免并发读/写